### PR TITLE
copr: implement several mysql time related types json

### LINKED ISSUE #13417: Support date/datetime/timestamp/time in JSON
## Development Task

The TiKV json implementation doesn't distinguish the `date/datetime/timestamp/time` type. The `json_type` of them returns the wrong value.

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What is changed and how it works?

Issue Number: Close #13417

What's Changed:

Add support for json date/datetime/timestamp/time types.

### Check List 

Tests

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- No code

### Release note 

```release-note
Fix the issue that TiKV doesn't distinguish the `DATETIME/DATE/TIMESTAMP/TIME` and `STRING` in json type.
```

### CODE REVIEW

**Comment by YangKeao** [comment]
@xiongjiwei PTAL

**Comment by YangKeao** [comment]
/cc @wshwsh12 @gengliqi @sticnarf

PTAL :heart:

**Review by xiongjiwei — APPROVED** [review]

File: components/tidb_query_datatype/src/codec/mysql/json/binary.rs [hunk 1]
```diff
@@ -88,6 +88,13 @@ impl<'a> JsonRef<'a> {
                     &self.value()[val_offset..val_offset + opaque_bytes_len as usize + len_len + 1],
                 )
             }
+            JsonType::Date | JsonType::Datetime | JsonType::Timestamp => {
```
Comment (sticnarf) [inline components/tidb_query_datatype/src/codec/mysql/json/binary.rs:91]: Is there any test case against these changes (including others in the `mysql/json` directory)

Comment (YangKeao) [reply components/tidb_query_datatype/src/codec/mysql/json/binary.rs:91]: Oops. It seems that the whole `get_entry` and `get_elem` doesn't have unit test. I'll add some test for them.

Comment (YangKeao) [reply components/tidb_query_datatype/src/codec/mysql/json/binary.rs:91]: Added!

**Review by sticnarf — APPROVED** [review]
Better to also add test cases for comparison and unquote

**Comment by YangKeao** [comment]
> Better to also add test cases for comparison and unquote

Added :smiley:

**Review by gengliqi — APPROVED** [review]

**Pull request merged**